### ISSUE TITLE
Clean up unit tests

### ISSUE THREAD

**Issue by stevemessick** [flutter/flutter-intellij#5999]
Running tests with 2022.1 causes a number of warnings to be generated. We need to update the test code to stop using deprecated APIs. Also, check that test failures still cause the process to exit with failed status.

**Comment by stevemessick** [comment]
Unfortunately, this needs to be deferred. Some of the deprecated methods are being replaced by new APIs that depend on classes not available to current stable and beta channels of Android Studio.

**Comment by stevemessick** [comment]
Android Studio is now based on 2021.2, so it is going to be a while before we can do this. I'll just keep moving it to the furthest milestone so we don't forget.

**Comment by stevemessick** [comment]
Fixed in #6386
